### PR TITLE
DAOS-9173 ofi: revert OFI patch and apply Mercury workaround

### PR DESCRIPTION
**Pull request by johannlombardi**
Due to some issues hit during soak (see DAOS-9195 for more info).
Pin both mercury and libfabric version to make sure to pick
the correct version.

PR-repos: libfabric@PR-49

Signed-off-by: Johann Lombardi <johann.lombardi@intel.com>

### CODE REVIEW

**Review by liuxuezhao — DISMISSED** [review]

**Comment by liuxuezhao** [comment]
the rpm build failed for "[2021-12-13T08:22:18.351Z] Error: No Package found for libfabric-devel = 1.14.0~rc3-1"

**Comment by johannlombardi** [comment]
I have just rebuilt 1.14.0~rc3-1 in the CI. Let me try to retrigger it.

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -30,7 +30,7 @@ BuildRequires: python36-scons >= 2.4
 %else
 BuildRequires: scons >= 2.4
 %endif
-BuildRequires: libfabric-devel >= %{libfabric_version}
+BuildRequires: libfabric-devel = %{libfabric_version}
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:33]: This (`=`) is really undesirable for similar reasons as below.

File: utils/rpms/daos.spec [hunk 2]
```diff
@@ -168,7 +168,7 @@ Requires: libpmemobj = 1.11.0-3%{?dist}
 Requires: mercury = %{mercury_version}
 Requires(post): /sbin/ldconfig
 Requires(postun): /sbin/ldconfig
-Requires: libfabric >= %{libfabric_version}
+Requires: libfabric = %{libfabric_version}
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:171]: This (`=`) is really undesirable (and unnecessary).  This restricts our ability to release a libfabric update in the future without having to go through a whole release process (i.e. package rebuild, QA, SDL, etc.) cycle for daos.  If we leave this as `>` and find a need in the future for a new libfabric, we can release just the new libfabric (after internal verification with the already released version of daos) and not have to go through the whole release process for a new daos that does nothing new except change it's libfabric requirement.

File: utils/rpms/daos.spec [hunk 3]
```diff
@@ -179,7 +179,7 @@ This is the package needed to run a DAOS server
 Summary: The DAOS client
 Requires: %{name}%{?_isa} = %{version}-%{release}
 Requires: mercury = %{mercury_version}
-Requires: libfabric >= %{libfabric_version}
+Requires: libfabric = %{libfabric_version}
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:182]: Ditto.

File: debian/control [hunk 4]
```diff
@@ -166,7 +166,7 @@ Section: net
 Architecture: any
 Multi-Arch: same
 Depends: ${shlibs:Depends}, ${misc:Depends}, openmpi-bin,
-         ipmctl (>02.00.00.3816), libfabric (>= 1.14.0~rc3-2)
+         ipmctl (>02.00.00.3816), libfabric (= 1.14.0~rc3-3)
```
Comment (brianjmurrell) [inline debian/control:169]: See comments below about `=` vs. `>=`.

File: debian/control [hunk 5]
```diff
@@ -147,7 +147,7 @@ Package: daos-client
 Section: net
 Architecture: any
 Multi-Arch: same
-Depends: ${shlibs:Depends}, ${misc:Depends}, openmpi-bin, libfabric (>= 1.14.0~rc3-2)
+Depends: ${shlibs:Depends}, ${misc:Depends}, openmpi-bin, libfabric (= 1.14.0~rc3-3)
```
Comment (brianjmurrell) [inline debian/control:150]: See comments below about `=` vs. `>=`.

File: debian/control [hunk 6]
```diff
@@ -5,7 +5,7 @@ Maintainer: daos-stack <daos@daos.groups.io>
 Build-Depends: debhelper (>= 10),
                dh-python,
                libcmocka-dev,
-               libfabric-dev (>= 1.14.0~rc3-2),
+               libfabric-dev (= 1.14.0~rc3-3),
```
Comment (brianjmurrell) [inline debian/control:8]: See comments below about `=` vs. `>=`.

**Review by frostedcmos — APPROVED** [review]

**Review by liuxuezhao — APPROVED** [review]

**Pull request closed without merging**